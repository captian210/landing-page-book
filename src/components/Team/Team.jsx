import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Social from "../Social";
import Card from "../Card";

class Team extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      showImage,
      showName,
      showJob,
      showSummary,
      showBorder,
      socials,
      socialGray,
      members,
      imageCircle,
      contentAlign,
      imageBorder
    } = this.props;
    return (
      <div className="Team row">
        {
          members.map((member, index) => {
            return (
              <div className={`col-12 col-md-4 text-${contentAlign} mb-4`} key={index}>
                <Card
                  imageCircle={imageCircle}
                  subTitle={showJob ? member.job : ""}
                  title={showName ? member.name : ""}
                  summary={showSummary ? member.summary : ""}
                  showBorder={showBorder}
                  contentAlign={contentAlign}
                  imageBorder={imageBorder}
                  image={showImage ? member.image : ""}>
                  {
                    socials.map((social, index) => {
                      return <Social key={index} url={member.profile} type={social} gray={socialGray}/>
                    })
                  }
                </Card>
              </div>
            )
          })
        }
      </div>
    )
  }
}

Team.propTypes = {
  showBorder: PropTypes.bool,
  showImage: PropTypes.bool,
  showName: PropTypes.bool,
  showJob: PropTypes.bool,
  showSummary: PropTypes.bool,
  imageCircle: PropTypes.bool,
  imageBorder: PropTypes.bool,
  socialGray: PropTypes.bool,
  contentAlign: PropTypes.string,
  socials: PropTypes.array,
  members: PropTypes.array
};
Team.defaultProps = {
  showBorder: true,
  showImage: true,
  showName: true,
  showJob: true,
  showSummary: true,
  imageCircle: true,
  imageBorder: false,
  socialGray: false,
  contentAlign: "center",
  socials: [],
  members: []
};

export default Team;
