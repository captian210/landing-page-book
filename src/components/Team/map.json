{"description":"","displayName":"Team","methods":[],"props":{"showBorder":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}},"showImage":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}},"showName":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}},"showJob":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}},"showSummary":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}},"imageCircle":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}},"imageBorder":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"false","computed":false}},"socialGray":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"false","computed":false}},"shadow":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"false","computed":false}},"imageShadow":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"false","computed":false}},"topColor":{"type":{"name":"string"},"required":false,"description":"","defaultValue":{"value":"'transparent'","computed":false}},"contentAlign":{"type":{"name":"string"},"required":false,"description":"","defaultValue":{"value":"'center'","computed":false}},"columnClassName":{"type":{"name":"string"},"required":false,"description":"","defaultValue":{"value":"'col-12 col-sm- col-md'","computed":false}},"members":{"type":{"name":"arrayOf","value":{"name":"shape","value":{"image":{"name":"string","required":false},"name":{"name":"string","required":false},"job":{"name":"string","required":false},"summary":{"name":"string","required":false},"profile":{"name":"arrayOf","value":{"name":"shape","value":{"social":{"name":"string","required":false},"url":{"name":"string","required":false}}},"required":false}}}},"required":false,"description":"","defaultValue":{"value":"[]","computed":false}}},"filename":"/components/Team/Team.jsx"}