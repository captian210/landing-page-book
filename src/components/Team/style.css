:root {
  --Card-borderRadius: 0px;
  --Card__Summary-fontSize: 0.8rem;
  --Card__Summary-color: #000;
  --Card__Summary-backgroundColor: #fff;
  --Card__Title-fontSize: 1.2rem;
  --Card__Title-color: #000;
  --Card__Subtitle-fontSize: 0.9rem;
  --Card__Subtitle-color: #b5b5b5; }

.Card {
  background-color: transparent;
  border-radius: var(--Card-borderRadius); }
  .Card .Card__Summary {
    font-size: var(--Card__Summary-fontSize);
    color: var(--Card__Summary-color);
    background-color: var(--Card__Summary-backgroundColor); }
  .Card .Card__Title {
    font-size: var(--Card__Title-fontSize);
    color: var(--Card__Title-color); }
  .Card .Card__Subtitle {
    font-size: var(--Card__Subtitle-fontSize);
    color: var(--Card__Subtitle-color); }

.SocialMediaShareButton {
  display: inline-block;
  cursor: pointer; }
  .SocialMediaShareButton div {
    width: 35px !important;
    height: 35px !important; }
  .SocialMediaShareButton svg {
    width: 30px;
    height: 30px; }

.SocialMediaShareButton--grayScale svg {
  filter: grayscale(1); }
