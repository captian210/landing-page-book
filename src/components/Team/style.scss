.Team{
  .Team__Name{
    font-size: $Team__Name-fontSize;
    color: $Team__Name-color;
  }
  .Team__Job{
    font-size: $Team__Job-fontSize;
    color: $Team__Job-color;
  }
  .Team__Summary{
    font-size: $Team__Summary-fontSize;
    color: $Team__Summary-color;
  }
  .Team__Card{
    border-radius: $Team__Card-borderRadius;
  }
}