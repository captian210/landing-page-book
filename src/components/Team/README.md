# Team

[![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)

## Overview
`@front10/landing-page-book/Team` is used to show a team.

## How to use
1- Importing bootstrap style

```js
import "bootstrap/dist/css/bootstrap.min.css";
```
2- Add following content to import the component:

```js
import Team from "@front10/landing-page-book/dist/components/Team";
import "@front10/landing-page-book/dist/components/Team/style.css";
```

> Note: For including all components styles once you can use [a theme](https://github.com/front10/landing-page-book/wiki/Theming).

3- Put this code into jsx page:
```html
<Team members={members} />
```

## Example
```js
const members = [{
	image: "https://randomuser.me/api/portraits/men/11.jpg",
	name: "Sander Coleman",
	job: "CEO",
	summary: "Graduated in 2002, from Harvard University, he participated for 3 years in doctors without borders.",
	profile: [
			{
				social: 'facebook',
				url: 'https://www.facebok.com/front10'
			},
			{
				social: 'twitter',
				url: 'https://twitter.com/landing-page-book'
			}
    ]
}, {
	image: "https://randomuser.me/api/portraits/women/2.jpg",
	name: "Rachel Lee",
	job: "Developer",
	summary: "Graduated in 2002, from Harvard University, he participated for 3 years in doctors without borders.",
	profile: [
			{
				social: 'facebook',
				url: 'https://www.facebook.com/front10'
			},
			{
				social: 'linkedin',
				url: 'https://www.linkedin.com/landing-page-book'
			}
    ]
}];
```

## Properties:

| </br>Name   | </br>Type | </br>Summary                                                                                 | 
| ------------| - | ------------------------------------------------------------------------------------------------------ |
| showImage      | `Boolean` | Show or hide image. Default `true` |
| showName      | `Boolean` | Show or hide name. Default `true` |
| showJob      | `Boolean` | Show or hide job. Default `true` |
| showSummary      | `Boolean` | Show or hide summary. Default `true` |
| showBorder      | `Boolean` | Show or hide card border. Default `true` |
| imageCircle      | `Boolean` | Establishes if image is circled. Default `true` |
| imageBorder      | `Boolean` | Establishes if image is with border. Default `false` |
| socialGray      | `Boolean` | Establishes if social icon is gray scale. Default `false` |
| contentAlign      | `String` | Align of a component content. Default `"center"`, can be `left` and `right` too|
| members      | `Array` | Array of members to show. Default `[]`, see example section.|
