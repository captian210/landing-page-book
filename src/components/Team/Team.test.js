import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Team from './Team';

configure({ adapter: new Adapter() });

describe('Team', () => {
  let props;
  let mounted;
  const team = () => {
    if (!mounted) {
      mounted = shallow(<Team {...props} />);
    }
    return mounted;
  };
  describe('Team', () => {
    beforeEach(() => {
      props = {
        showBorder: true,
        showImage: true,
        showName: true,
        showJob: true,
        showSummary: true,
        imageCircle: true,
        imageBorder: false,
        socialGray: false,
        shadow: false,
        imageShadow: false,
        topColor: 'transparent',
        contentAlign: 'center',
        members: [
          {
            image: '/images/team/ninja.svg',
            job: 'Ninja',
            name: 'Leonardo',
            profile: [
              {
                social: 'facebook',
                url: 'https://www.facebook.com/JavaScriptTeam/'
              },
              {
                social: 'twitter',
                url: 'https://twitter.com/front10hello?lang=es'
              }
            ],
            summary:
              'Leonardo, nickname Leo, is a fictional character and one of the four main characters in the Teenage Mutant Ninja Turtles comics and related media.'
          },
          {
            image: '/images/team/bohemian.svg',
            job: 'Painter',
            name: 'Van Gogh',
            profile: [
              {
                social: 'facebook',
                url: 'https://www.facebook.com/JavaScriptTeam/'
              },
              {
                social: 'email',
                url: 'https://front10.com'
              }
            ]
          }
        ]
      };
      mounted = undefined;
    });

    it('should render', () => {
      expect(shallow(<Team {...props} />)).toMatchSnapshot();
    });

    it('always renders a div', () => {
      const divs = team().find('div');

      expect(divs.length).toBeGreaterThan(0);
    });
  });
});
