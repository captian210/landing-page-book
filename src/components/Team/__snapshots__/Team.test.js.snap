// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Team should render 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Team
    contentAlign="center"
    imageBorder={false}
    imageCircle={true}
    members={Array []}
    showBorder={true}
    showImage={true}
    showJob={true}
    showName={true}
    showSummary={true}
    socialGray={false}
    socials={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": Team {
      "_reactInternalFiber": Se {
        "alternate": null,
        "child": Se {
          "alternate": null,
          "child": null,
          "childExpirationTime": 0,
          "effectTag": 0,
          "expirationTime": 0,
          "firstContextDependency": null,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": Array [],
            "className": "Team row",
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "children": Array [],
            "className": "Team row",
          },
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": <div
            class="Team row"
          />,
          "tag": 7,
          "type": "div",
          "updateQueue": null,
        },
        "childExpirationTime": 0,
        "effectTag": 1,
        "expirationTime": 0,
        "firstContextDependency": null,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "contentAlign": "center",
          "imageBorder": false,
          "imageCircle": true,
          "members": Array [],
          "showBorder": true,
          "showImage": true,
          "showJob": true,
          "showName": true,
          "showSummary": true,
          "socialGray": false,
          "socials": Array [],
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "contentAlign": "center",
          "imageBorder": false,
          "imageCircle": true,
          "members": Array [],
          "showBorder": true,
          "showImage": true,
          "showJob": true,
          "showName": true,
          "showSummary": true,
          "socialGray": false,
          "socials": Array [],
        },
        "ref": null,
        "return": Se {
          "alternate": null,
          "child": [Circular],
          "childExpirationTime": 0,
          "effectTag": 1,
          "expirationTime": 0,
          "firstContextDependency": null,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "Component": [Function],
            "context": null,
            "props": Object {
              "contentAlign": "center",
              "imageBorder": false,
              "imageCircle": true,
              "members": Array [],
              "showBorder": true,
              "showImage": true,
              "showJob": true,
              "showName": true,
              "showSummary": true,
              "socialGray": false,
              "socials": Array [],
            },
          },
          "memoizedState": Object {
            "context": null,
            "mount": true,
            "props": Object {
              "contentAlign": "center",
              "imageBorder": false,
              "imageCircle": true,
              "members": Array [],
              "showBorder": true,
              "showImage": true,
              "showJob": true,
              "showName": true,
              "showSummary": true,
              "socialGray": false,
              "socials": Array [],
            },
          },
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "Component": [Function],
            "context": null,
            "props": Object {
              "contentAlign": "center",
              "imageBorder": false,
              "imageCircle": true,
              "members": Array [],
              "showBorder": true,
              "showImage": true,
              "showJob": true,
              "showName": true,
              "showSummary": true,
              "socialGray": false,
              "socials": Array [],
            },
          },
          "ref": null,
          "return": Se {
            "alternate": Se {
              "alternate": [Circular],
              "child": null,
              "childExpirationTime": 0,
              "effectTag": 0,
              "expirationTime": 1,
              "firstContextDependency": null,
              "firstEffect": null,
              "index": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": null,
              "memoizedState": null,
              "mode": 0,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <div
                    class="Team row"
                  />
                </div>,
                "context": Object {},
                "current": [Circular],
                "didError": false,
                "earliestPendingTime": 0,
                "earliestSuspendedTime": 0,
                "expirationTime": 0,
                "finishedWork": null,
                "firstBatch": null,
                "hydrate": false,
                "latestPendingTime": 0,
                "latestPingedTime": 0,
                "latestSuspendedTime": 0,
                "nextExpirationTimeToWorkOn": 0,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingCommitExpirationTime": 0,
                "pendingContext": null,
                "timeoutHandle": -1,
              },
              "tag": 5,
              "type": null,
              "updateQueue": Object {
                "baseState": null,
                "firstCapturedEffect": null,
                "firstCapturedUpdate": null,
                "firstEffect": null,
                "firstUpdate": Object {
                  "callback": null,
                  "expirationTime": 1,
                  "next": null,
                  "nextEffect": null,
                  "payload": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={
                        Object {
                          "contentAlign": "center",
                          "imageBorder": false,
                          "imageCircle": true,
                          "members": Array [],
                          "showBorder": true,
                          "showImage": true,
                          "showJob": true,
                          "showName": true,
                          "showSummary": true,
                          "socialGray": false,
                          "socials": Array [],
                        }
                      }
                    />,
                  },
                  "tag": 0,
                },
                "lastCapturedEffect": null,
                "lastCapturedUpdate": null,
                "lastEffect": null,
                "lastUpdate": Object {
                  "callback": null,
                  "expirationTime": 1,
                  "next": null,
                  "nextEffect": null,
                  "payload": Object {
                    "element": <WrapperComponent
                      Component={[Function]}
                      context={null}
                      props={
                        Object {
                          "contentAlign": "center",
                          "imageBorder": false,
                          "imageCircle": true,
                          "members": Array [],
                          "showBorder": true,
                          "showImage": true,
                          "showJob": true,
                          "showName": true,
                          "showSummary": true,
                          "socialGray": false,
                          "socials": Array [],
                        }
                      }
                    />,
                  },
                  "tag": 0,
                },
              },
            },
            "child": [Circular],
            "childExpirationTime": 0,
            "effectTag": 32,
            "expirationTime": 0,
            "firstContextDependency": null,
            "firstEffect": [Circular],
            "index": 0,
            "key": null,
            "lastEffect": [Circular],
            "memoizedProps": null,
            "memoizedState": Object {
              "element": <WrapperComponent
                Component={[Function]}
                context={null}
                props={
                  Object {
                    "contentAlign": "center",
                    "imageBorder": false,
                    "imageCircle": true,
                    "members": Array [],
                    "showBorder": true,
                    "showImage": true,
                    "showJob": true,
                    "showName": true,
                    "showSummary": true,
                    "socialGray": false,
                    "socials": Array [],
                  }
                }
              />,
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": null,
            "sibling": null,
            "stateNode": Object {
              "containerInfo": <div>
                <div
                  class="Team row"
                />
              </div>,
              "context": Object {},
              "current": [Circular],
              "didError": false,
              "earliestPendingTime": 0,
              "earliestSuspendedTime": 0,
              "expirationTime": 0,
              "finishedWork": null,
              "firstBatch": null,
              "hydrate": false,
              "latestPendingTime": 0,
              "latestPingedTime": 0,
              "latestSuspendedTime": 0,
              "nextExpirationTimeToWorkOn": 0,
              "nextScheduledRoot": null,
              "pendingChildren": null,
              "pendingCommitExpirationTime": 0,
              "pendingContext": null,
              "timeoutHandle": -1,
            },
            "tag": 5,
            "type": null,
            "updateQueue": Object {
              "baseState": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "contentAlign": "center",
                      "imageBorder": false,
                      "imageCircle": true,
                      "members": Array [],
                      "showBorder": true,
                      "showImage": true,
                      "showJob": true,
                      "showName": true,
                      "showSummary": true,
                      "socialGray": false,
                      "socials": Array [],
                    }
                  }
                />,
              },
              "firstCapturedEffect": null,
              "firstCapturedUpdate": null,
              "firstEffect": null,
              "firstUpdate": null,
              "lastCapturedEffect": null,
              "lastCapturedUpdate": null,
              "lastEffect": null,
              "lastUpdate": null,
            },
          },
          "sibling": null,
          "stateNode": WrapperComponent {
            "_reactInternalFiber": [Circular],
            "context": Object {},
            "props": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "contentAlign": "center",
                "imageBorder": false,
                "imageCircle": true,
                "members": Array [],
                "showBorder": true,
                "showImage": true,
                "showJob": true,
                "showName": true,
                "showSummary": true,
                "socialGray": false,
                "socials": Array [],
              },
            },
            "refs": Object {},
            "state": Object {
              "context": null,
              "mount": true,
              "props": Object {
                "contentAlign": "center",
                "imageBorder": false,
                "imageCircle": true,
                "members": Array [],
                "showBorder": true,
                "showImage": true,
                "showJob": true,
                "showName": true,
                "showSummary": true,
                "socialGray": false,
                "socials": Array [],
              },
            },
            "updater": Object {
              "enqueueForceUpdate": [Function],
              "enqueueReplaceState": [Function],
              "enqueueSetState": [Function],
              "isMounted": [Function],
            },
          },
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "sibling": null,
        "stateNode": [Circular],
        "tag": 2,
        "type": [Function],
        "updateQueue": null,
      },
      "context": Object {},
      "props": Object {
        "contentAlign": "center",
        "imageBorder": false,
        "imageCircle": true,
        "members": Array [],
        "showBorder": true,
        "showImage": true,
        "showJob": true,
        "showName": true,
        "showSummary": true,
        "socialGray": false,
        "socials": Array [],
      },
      "refs": Object {},
      "state": null,
      "updater": Object {
        "enqueueForceUpdate": [Function],
        "enqueueReplaceState": [Function],
        "enqueueSetState": [Function],
        "isMounted": [Function],
      },
    },
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "contentAlign": "center",
      "imageBorder": false,
      "imageCircle": true,
      "members": Array [],
      "showBorder": true,
      "showImage": true,
      "showJob": true,
      "showName": true,
      "showSummary": true,
      "socialGray": false,
      "socials": Array [],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="Team row"
      />,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "Team row",
      },
      "ref": null,
      "rendered": Array [
        Array [],
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": Team {
        "_reactInternalFiber": Se {
          "alternate": null,
          "child": Se {
            "alternate": null,
            "child": null,
            "childExpirationTime": 0,
            "effectTag": 0,
            "expirationTime": 0,
            "firstContextDependency": null,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "children": Array [],
              "className": "Team row",
            },
            "memoizedState": null,
            "mode": 0,
            "nextEffect": null,
            "pendingProps": Object {
              "children": Array [],
              "className": "Team row",
            },
            "ref": null,
            "return": [Circular],
            "sibling": null,
            "stateNode": <div
              class="Team row"
            />,
            "tag": 7,
            "type": "div",
            "updateQueue": null,
          },
          "childExpirationTime": 0,
          "effectTag": 1,
          "expirationTime": 0,
          "firstContextDependency": null,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "contentAlign": "center",
            "imageBorder": false,
            "imageCircle": true,
            "members": Array [],
            "showBorder": true,
            "showImage": true,
            "showJob": true,
            "showName": true,
            "showSummary": true,
            "socialGray": false,
            "socials": Array [],
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "contentAlign": "center",
            "imageBorder": false,
            "imageCircle": true,
            "members": Array [],
            "showBorder": true,
            "showImage": true,
            "showJob": true,
            "showName": true,
            "showSummary": true,
            "socialGray": false,
            "socials": Array [],
          },
          "ref": null,
          "return": Se {
            "alternate": null,
            "child": [Circular],
            "childExpirationTime": 0,
            "effectTag": 1,
            "expirationTime": 0,
            "firstContextDependency": null,
            "firstEffect": null,
            "index": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "contentAlign": "center",
                "imageBorder": false,
                "imageCircle": true,
                "members": Array [],
                "showBorder": true,
                "showImage": true,
                "showJob": true,
                "showName": true,
                "showSummary": true,
                "socialGray": false,
                "socials": Array [],
              },
            },
            "memoizedState": Object {
              "context": null,
              "mount": true,
              "props": Object {
                "contentAlign": "center",
                "imageBorder": false,
                "imageCircle": true,
                "members": Array [],
                "showBorder": true,
                "showImage": true,
                "showJob": true,
                "showName": true,
                "showSummary": true,
                "socialGray": false,
                "socials": Array [],
              },
            },
            "mode": 0,
            "nextEffect": null,
            "pendingProps": Object {
              "Component": [Function],
              "context": null,
              "props": Object {
                "contentAlign": "center",
                "imageBorder": false,
                "imageCircle": true,
                "members": Array [],
                "showBorder": true,
                "showImage": true,
                "showJob": true,
                "showName": true,
                "showSummary": true,
                "socialGray": false,
                "socials": Array [],
              },
            },
            "ref": null,
            "return": Se {
              "alternate": Se {
                "alternate": [Circular],
                "child": null,
                "childExpirationTime": 0,
                "effectTag": 0,
                "expirationTime": 1,
                "firstContextDependency": null,
                "firstEffect": null,
                "index": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": null,
                "memoizedState": null,
                "mode": 0,
                "nextEffect": null,
                "pendingProps": null,
                "ref": null,
                "return": null,
                "sibling": null,
                "stateNode": Object {
                  "containerInfo": <div>
                    <div
                      class="Team row"
                    />
                  </div>,
                  "context": Object {},
                  "current": [Circular],
                  "didError": false,
                  "earliestPendingTime": 0,
                  "earliestSuspendedTime": 0,
                  "expirationTime": 0,
                  "finishedWork": null,
                  "firstBatch": null,
                  "hydrate": false,
                  "latestPendingTime": 0,
                  "latestPingedTime": 0,
                  "latestSuspendedTime": 0,
                  "nextExpirationTimeToWorkOn": 0,
                  "nextScheduledRoot": null,
                  "pendingChildren": null,
                  "pendingCommitExpirationTime": 0,
                  "pendingContext": null,
                  "timeoutHandle": -1,
                },
                "tag": 5,
                "type": null,
                "updateQueue": Object {
                  "baseState": null,
                  "firstCapturedEffect": null,
                  "firstCapturedUpdate": null,
                  "firstEffect": null,
                  "firstUpdate": Object {
                    "callback": null,
                    "expirationTime": 1,
                    "next": null,
                    "nextEffect": null,
                    "payload": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={
                          Object {
                            "contentAlign": "center",
                            "imageBorder": false,
                            "imageCircle": true,
                            "members": Array [],
                            "showBorder": true,
                            "showImage": true,
                            "showJob": true,
                            "showName": true,
                            "showSummary": true,
                            "socialGray": false,
                            "socials": Array [],
                          }
                        }
                      />,
                    },
                    "tag": 0,
                  },
                  "lastCapturedEffect": null,
                  "lastCapturedUpdate": null,
                  "lastEffect": null,
                  "lastUpdate": Object {
                    "callback": null,
                    "expirationTime": 1,
                    "next": null,
                    "nextEffect": null,
                    "payload": Object {
                      "element": <WrapperComponent
                        Component={[Function]}
                        context={null}
                        props={
                          Object {
                            "contentAlign": "center",
                            "imageBorder": false,
                            "imageCircle": true,
                            "members": Array [],
                            "showBorder": true,
                            "showImage": true,
                            "showJob": true,
                            "showName": true,
                            "showSummary": true,
                            "socialGray": false,
                            "socials": Array [],
                          }
                        }
                      />,
                    },
                    "tag": 0,
                  },
                },
              },
              "child": [Circular],
              "childExpirationTime": 0,
              "effectTag": 32,
              "expirationTime": 0,
              "firstContextDependency": null,
              "firstEffect": [Circular],
              "index": 0,
              "key": null,
              "lastEffect": [Circular],
              "memoizedProps": null,
              "memoizedState": Object {
                "element": <WrapperComponent
                  Component={[Function]}
                  context={null}
                  props={
                    Object {
                      "contentAlign": "center",
                      "imageBorder": false,
                      "imageCircle": true,
                      "members": Array [],
                      "showBorder": true,
                      "showImage": true,
                      "showJob": true,
                      "showName": true,
                      "showSummary": true,
                      "socialGray": false,
                      "socials": Array [],
                    }
                  }
                />,
              },
              "mode": 0,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": null,
              "sibling": null,
              "stateNode": Object {
                "containerInfo": <div>
                  <div
                    class="Team row"
                  />
                </div>,
                "context": Object {},
                "current": [Circular],
                "didError": false,
                "earliestPendingTime": 0,
                "earliestSuspendedTime": 0,
                "expirationTime": 0,
                "finishedWork": null,
                "firstBatch": null,
                "hydrate": false,
                "latestPendingTime": 0,
                "latestPingedTime": 0,
                "latestSuspendedTime": 0,
                "nextExpirationTimeToWorkOn": 0,
                "nextScheduledRoot": null,
                "pendingChildren": null,
                "pendingCommitExpirationTime": 0,
                "pendingContext": null,
                "timeoutHandle": -1,
              },
              "tag": 5,
              "type": null,
              "updateQueue": Object {
                "baseState": Object {
                  "element": <WrapperComponent
                    Component={[Function]}
                    context={null}
                    props={
                      Object {
                        "contentAlign": "center",
                        "imageBorder": false,
                        "imageCircle": true,
                        "members": Array [],
                        "showBorder": true,
                        "showImage": true,
                        "showJob": true,
                        "showName": true,
                        "showSummary": true,
                        "socialGray": false,
                        "socials": Array [],
                      }
                    }
                  />,
                },
                "firstCapturedEffect": null,
                "firstCapturedUpdate": null,
                "firstEffect": null,
                "firstUpdate": null,
                "lastCapturedEffect": null,
                "lastCapturedUpdate": null,
                "lastEffect": null,
                "lastUpdate": null,
              },
            },
            "sibling": null,
            "stateNode": WrapperComponent {
              "_reactInternalFiber": [Circular],
              "context": Object {},
              "props": Object {
                "Component": [Function],
                "context": null,
                "props": Object {
                  "contentAlign": "center",
                  "imageBorder": false,
                  "imageCircle": true,
                  "members": Array [],
                  "showBorder": true,
                  "showImage": true,
                  "showJob": true,
                  "showName": true,
                  "showSummary": true,
                  "socialGray": false,
                  "socials": Array [],
                },
              },
              "refs": Object {},
              "state": Object {
                "context": null,
                "mount": true,
                "props": Object {
                  "contentAlign": "center",
                  "imageBorder": false,
                  "imageCircle": true,
                  "members": Array [],
                  "showBorder": true,
                  "showImage": true,
                  "showJob": true,
                  "showName": true,
                  "showSummary": true,
                  "socialGray": false,
                  "socials": Array [],
                },
              },
              "updater": Object {
                "enqueueForceUpdate": [Function],
                "enqueueReplaceState": [Function],
                "enqueueSetState": [Function],
                "isMounted": [Function],
              },
            },
            "tag": 2,
            "type": [Function],
            "updateQueue": null,
          },
          "sibling": null,
          "stateNode": [Circular],
          "tag": 2,
          "type": [Function],
          "updateQueue": null,
        },
        "context": Object {},
        "props": Object {
          "contentAlign": "center",
          "imageBorder": false,
          "imageCircle": true,
          "members": Array [],
          "showBorder": true,
          "showImage": true,
          "showJob": true,
          "showName": true,
          "showSummary": true,
          "socialGray": false,
          "socials": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
        },
      },
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "contentAlign": "center",
        "imageBorder": false,
        "imageCircle": true,
        "members": Array [],
        "showBorder": true,
        "showImage": true,
        "showJob": true,
        "showName": true,
        "showSummary": true,
        "socialGray": false,
        "socials": Array [],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="Team row"
        />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "Team row",
        },
        "ref": null,
        "rendered": Array [
          Array [],
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
